core/lib/patternlab.js

```diff
@@ -35,18 +36,43 @@ function buildPatternData(dataFilesPath, fsDep) {
 
 // GTP: these two diveSync pattern processors factored out so they can be reused
 // from unit tests to reduce code dupe!
-function processAllPatternsIterative(pattern_assembler, patterns_dir, patternlab) {
-  diveSync(
-    patterns_dir,
-    function (err, file) {
-      //log any errors
-      if (err) {
-        console.log(err);
-        return;
-      }
-      pattern_assembler.process_pattern_iterative(path.relative(patterns_dir, file), patternlab);
-    }
-  );
+function processAllPatternsIterative(pattern_assembler, patterns_dir, patternlab){
+  const async = require('async');
```

Comment:
Yup. I need to take that out. I'm not currently using async for anything. I thought I would need it, but it turns out that I don't have a need for it so far. And in any case, it's not very Promise-y.